infra/modules/services/diagnosticsettings.bicep

```diff
@@ -0,0 +1,304 @@
+  parent: sqlserver
+  name: 'Database001'
+}
+
+resource eventhubNamespace 'Microsoft.EventHub/namespaces@2021-06-01-preview' existing = {
+  name: eventhubnamespaceName
+}
+
+resource streamanalyticsjob001 'Microsoft.StreamAnalytics/streamingjobs@2020-03-01' existing = {
+  name: streamanalyticsName
+}
+
+resource diagnosticSetting001 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  scope: synapseworkspace
+  name: 'diagnostic-${synapseworkspace.name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'SynapseRbacOperations'
+        enabled: true
+      }
+      {
+        category: 'GatewayApiRequests'
+        enabled: true
+      }
+      {
+        category: 'BuiltinSqlReqsEnded'
+        enabled: true
+      }
+      {
+        category: 'IntegrationPipelineRuns'
+        enabled: true
+      }
+      {
+        category: 'IntegrationActivityRuns'
+        enabled: true
+      }
+      {
+        category: 'IntegrationTriggerRuns'
+        enabled: true
+      }
+    ]
+  }
+}
+
+resource diagnosticSetting2 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for i in range(0, synapseSqlPoolsCount): {
+  scope: synapsesqlpool[i]
+  name: 'diagnostic-${synapseworkspace.name}-${synapsesqlpool[i].name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'SqlRequests'
+        enabled: true
+      }
+      {
+        category: 'RequestSteps'
+        enabled: true
+      }
+      {
+        category: 'ExecRequests'
+        enabled: true
+      }
+      {
+        category: 'DmsWorkers'
+        enabled: true
+      }
+      {
+        category: 'Waits'
+        enabled: true
+      }
+    ]
+  }
+}]
+
+resource diagnosticSetting3 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for i in range(0, synapseSparkPoolCount): {
+  scope: synapsebigdatapool[i]
+  name: 'diagnostic-${synapseworkspace.name}-${synapsebigdatapool[i].name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'BigDataPoolAppsEnded'
+        enabled: true
+      }
+    ]
+  }
+}]
+
+resource diagnosticSetting4 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  scope: cosmosdb
+  name: 'diagnostic-${cosmosdb.name}'
+  properties: {
+    workspaceId: logAnalyticsWorkspace.id
+    logs: [
+      {
+        category: 'DataPlaneRequests'
+        enabled: true
+      }
+      {
+        category: 'QueryRuntimeStatistics'
+        enabled: true
+      }
+      {
+        category: 'PartitionKeyStatistics'
+        enabled: true
+      }
+      {
+        category: 'PartitionKeyRUConsumption'
+        enabled: true
+      }
+      {
+        category: 'ControlPlaneRequests'
+        enabled: true
+      }
+    ]
+  }
+}
+
+resource diagnosticSetting5 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = {
+  scope: iothub
```

Comment:
Will this work if to set to false in params ?

Reply:
I get a similar error as when I set enableCosmos to false.